File: examples/ruby-1.9.3/http_post.rb
#!/usr/bin/env ruby

# Very basic Ruby script to post to a phant instance.
# This is unlikely to be very idiomatic; better examples
# welcome.

require 'net/http'

endpoint = 'https://data.sparkfun.com'
pubhash  = '6JJ4dEYoqNTwjyMVn0ZM'
privhash = 'WwwZVA7W5PiMG92Ym0p2'

data = { 'code' => 'ruby-1.9.3' }

puts "Who are you? "
data['who'] = gets

puts "Where are you at? "
data['where'] = gets

puts "What's your favorite animal? "
data['favorite_animal'] = gets

path = "#{endpoint}/input/#{pubhash}"
uri = URI(path)

req = Net::HTTP::Post.new(uri.path)
req.set_form_data(data)
req['Phant-Private-Key'] = privhash

res = Net::HTTP.start(uri.hostname, uri.port, :use_ssl => true) do |http|
  http.request(req)
end

puts res.body

File: examples/perl/http_post.pl
#!/usr/bin/env perl

=pod

=head1 NAME

http_post.pl - post some data to a Phant instance (like data.sparkfun.com)

=cut

use warnings;
use strict;
use LWP::UserAgent;
use HTTP::Request::Common qw(POST);

my $endpoint = 'https://data.sparkfun.com';
my $pubhash  = '6JJ4dEYoqNTwjyMVn0ZM';
my $privhash = 'WwwZVA7W5PiMG92Ym0p2';

my %data = (code => 'Perl');

print "Who are you? ";
$data{who} = <STDIN>;

print "Where are you at? ";
$data{where} = <STDIN>;

print "What's your favorite animal? ";
$data{favorite_animal} = <STDIN>;

my $ua = LWP::UserAgent->new;
$ua->agent('http_post.pl 1.0');

my $response = $ua->request(
  POST $endpoint . '/input/' . $pubhash,
       'Phant-Private-Key' => $privhash,
       Content => \%data
);

if ($response->is_success) {
  print $response->decoded_content;
} else {
  die $response->status_line . ' ' . $response->content;
}
